src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -223,17 +223,24 @@ public string ToString(string format, IFormatProvider provider)
 
         public static Complex Sin(Complex value)
         {
-            double a = value._real;
-            double b = value._imaginary;
-            return new Complex(Math.Sin(a) * Math.Cosh(b), Math.Cos(a) * Math.Sinh(b));
+            // We need both sinh and cosh of imaginary part. To avoid multiple calls to Math.Exp with the same value,
+            // we compute them both here from a single call to Math.Exp.
+            double p = Math.Exp(value._imaginary);
+            double q = 1.0 / p;
+            double sinh = (p - q) * 0.5;
+            double cosh = (p + q) * 0.5;
+            return new Complex(Math.Sin(value._real) * cosh, Math.Cos(value._real) * sinh);
+            // There is a small region where sinh and/or cosh (correctly) overflow, while sin and/or cos are small enough
```

Comment:
Addressed all these requests in the latest PR update. Just on this one I left a little more info in the code comment than I think you might have preferred, because I really want someone reading the comment to know the nature, not just the existence, of the limitation. I did include a specific example as you requested.